### PR TITLE
#163858430 user should be able to view and edit their profile

### PR DESCRIPTION
**Pull request by Hesbon5600**
#### What does this PR do?
* Implement user view and edit their profile

#### Description of Task to be completed?
* A user can view their profile
* A user can edit their profile
#### How should this be manually tested?
* Clone the github repository `git clone https://github.com/andela/ah-fulldeck-frontend` on your local machine.
* Change your working directory to the project folder `cd ah-fulldeck-frontend`
* Checkout to thebranch `ft-user-profile-163858430` using the command `git checkout ft-user-profile-163858430`
* Install the required dependencies using `npm install`
* Run the local development server using `npm start` and you should get the following message:
* Go to the link `/profile/<username>` where the `username` is your registered username

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#162948946](https://www.pivotaltracker.com/story/show/163858430)

#### Screenshots (if appropriate)
* User view their profile
screenshot 2019-02-13 at 16 42 24
*  User edit their profile
screenshot 2019-02-13 at 16 41 33
* Edited profile
screenshot 2019-02-13 at 18 35 57

### CODE REVIEW

File: src/actions/Profile/userProfileAction.jsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { GET_PROFILE, EDIT_PROFILE } from "../types";
+import axios from 'axios'
+export const getUserProfile = (username) => async dispatch => {
+  
+  const res = await axios.get(`https://ah-fulldeck-staging.herokuapp.com/api/v1/users/${username}`);
```
Comment (victorjambo) [inline src/actions/Profile/userProfileAction.jsx]: the api base url should be an environment variable

File: src/components/Users/UserProfileComponent.jsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      });
+    });
+  }
+  onEditClick = username => {};
+  render() {
+    if (!this.state.fetched) {
+      return <Loader />;
+    } 
+    if(this.props.profile.errors){
+      return(<NotFound/>)
```
Comment (eduhmik) [inline src/components/Users/UserProfileComponent.jsx:34]: Great implementation of a 404 page, I think most of us will reuse this component

**Review by eduhmik — DISMISSED** [review]

**Review by Nyakaru — DISMISSED** [review]
LGTM

**Review by JoshuaKodhe — APPROVED** [review]
LGTM

**Review by Nyakaru — APPROVED** [review]

**Review by charitymarani — APPROVED** [review]

**Pull request merged**